### ISSUE TITLE
RGB -> HSL issues

### ISSUE THREAD

**Issue by wilsonjord** [TurkeyMan/color#15]
```d
// passes
assert(convertColor!(RGB8)(convertColor!(XYZ!float)(RGB8(173, 156, 158))) == RGB8(173, 156, 158));
// fails
assert(convertColor!(RGB8)(convertColor!(HSL!float)(RGB8(173, 156, 158))) == RGB8(173, 156, 158));
```

I believe the degrees property in the hsx module has a problem:
```d
return (h < 0 ? 1 - h%1 : h%1) * 360;
```

Modulus result should have the same sign as the dividend, so given a negative hue angle, the modulus will be negative, so will end up being added to 1.

The following change results in both the asserts passing:
```d
return (h < 0 ? 1 + h%1 : h%1) * 360;
```

If I'm right, then I think the radians function may also be effected.

Thanks,
Jordan

**Comment by TurkeyMan** [comment]
Excellent catch! Thanks for spotting that!

**Comment by TurkeyMan** [comment]
Fixed in 0.0.9
